### PR TITLE
IdentityFetcher-rewrite part 69: Continue merging branch next into the rewrite

### PR DESCRIPTION
**Pull request by xor-freenet**
_Please merge with `--ff-only` into `issue-0003816-IdentityFetcher-rewrite`._
_This also contains the commits of the previous PRs to that branch._

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,31 +38,67 @@ cache:
...
+    # The gradlew binary which fred ships doesn't work on OpenJDK 7, need to use Travis' gradle there.
+    if [ "$TRAVIS_JDK_VERSION" = "openjdk7" ] ; then
+      rm ./gradlew &&
+      ln -s "$(which gradle)" ./gradlew
+    fi &&
+    if [[ "$TRAVIS_JDK_VERSION" = "openjdk9" || "$TRAVIS_JDK_VERSION" = "openjdk10" ]] ; then
+      # Workaround for Gradle failing on openjdk9 and above with:
+      #   sun.security.validator.ValidatorException: PKIX path building failed: [...]
+      # TODO: Code quality: Added 2018-09-26, remove after some time and see if it works without this.
+      USE_SYSTEM_CERTS=1 &&
```
Comment (ArneBab) [inline .travis.yml:66]: does this actually set the environment variable with the &&? Does this need export before the variable?

File: developer-documentation/changelogs/build0021.txt [hunk 2]
```diff
@@ -316,6 +334,22 @@ CHANGELOG about stuff only interesting for developers
   improvements over the heuristics of the one this release ships, go
   ahead, it should be easy :)
 
+- Code quality: Travis CI: Test against most recent Java versions (xor)
+
+  The cloud service now runs the unit tests periodically against:
+  - oraclejdk10 (new)
```
Comment (ArneBab) [inline developer-documentation/changelogs/build0021.txt:340]: remember to keep the period raw enough so this does not deplete all the GHA runs Freenet can do.

File: .travis.yml (hunk 1 above)
Comment (xor-freenet) [reply .travis.yml:66]: > Does this need export before the variable?

The variable is used in the `.travis.yml` file itself, merely for its own purposes, hence no export is needed.

> does this actually set the environment variable with the &&? 

The following prints "works" for me:
`bash -c 'testvar=works && echo $testvar'`

For completeness:
I don't remember if Travis strictly said it uses bash.

But this code will be converted to GHA by applying the Freetalk GHA code.
And that code explicitly declares the shell to be bash, so we're good.

File: developer-documentation/changelogs/build0021.txt (hunk 2 above)
Comment (xor-freenet) [reply developer-documentation/changelogs/build0021.txt:340]: > remember to keep the period raw enough so this does not deplete all the GHA runs Freenet can do.

See my email and reply to your other review here:
https://github.com/ArneBab/plugin-Freetalk-staging/pull/52#pullrequestreview-1048470514

**Pull request merged**